handlers/GameHandler.java

```diff
@@ -0,0 +1,68 @@
+package com.kbh.snl.handlers;
+
+import com.kbh.snl.models.Game;
+import com.kbh.snl.models.IntermediateObject;
+import com.kbh.snl.models.Ladder;
+import com.kbh.snl.models.Player;
+import com.kbh.snl.models.Snake;
+import java.util.Iterator;
+import java.util.Random;
+
+public class GameHandler {
+
+  private final Game game;
+  private final Random randomNumberGenerator;
+
+  public GameHandler(final Game game) {
+    this.game = game;
+    this.randomNumberGenerator = new Random();
+  }
+
+  public void rungame() {
+    while (game.getPlayersLeft().size() > 1) {
+      final Iterator<Player> playerIterator = game.getPlayersLeft().iterator();
+      executeNextStep(playerIterator);
+
+    }
+    System.out.println(String.format("%s wins the game", game.getSortedWinners()));
+  }
+
+  private void executeNextStep(Iterator<Player> playerIterator) {
+    while (playerIterator.hasNext()) {
+      final Player currentPlayer = playerIterator.next();
+      //throw dice
+      final int diceOutcome = this.throwdice();
+      //add current outcome to player position
+      int newPosition = currentPlayer.getCurrentPosition() + diceOutcome;
+      //if out of bounds then return else proceed
+      if (newPosition > game.getTotalCells()) {
+        return;
+      }
+      //check for ladders and snakes
+      if (game.getBoard().containsKey(newPosition)) {
+        final IntermediateObject intermediateObject = game.getBoard().get(newPosition);
```

Comment:
IntermediateObject name is not sufficient to understand it's purpose

Reply:
Can rename as Jumper? IntermediateObject is the base class extended by Snake and Ladder.